src/freenet/io/comm/MessageCore.java

```diff
@@ -616,9 +616,9 @@ public Map<String, Integer> getUnclaimedFIFOMessageCounts() {
 				String messageName = m.getSpec().getName();
 				Integer messageCount = messageCounts.get(messageName);
```

Comment:
This looks like a prime contender for `Map.merge()`…

Reply:
I will look into it another time. Please don't hold up the PR waiting for me.